Tests/Unit/MSFT_xIPAddress.Tests.ps1

```diff
@@ -119,102 +178,246 @@ InModuleScope MSFT_xIPAddress {
-            It 'should call Get-NetIPInterface once' {
-                Assert-MockCalled -CommandName Get-NetIPInterface
+            It 'should throw an error when greater than 32' {
+                $Splat = @{
+                    IPAddress = '192.168.0.1'
+                    InterfaceAlias = 'Ethernet'
+                    SubnetMask = 33
+                    AddressFamily = 'IPv4'
+                }
+                { Test-ResourceProperty @Splat } | Should Throw
+            }
+            It 'should throw an error when less than 0' {
+                $Splat = @{
+                    IPAddress = '192.168.0.1'
+                    InterfaceAlias = 'Ethernet'
+                    SubnetMask = -1
+                    AddressFamily = 'IPv4'
+                }
+                { Test-ResourceProperty @Splat } | Should Throw
             }
         }
 
-        Context 'invoking with -Apply switch' {
+        Context 'invoking with invalid IPv6 subnet mask' {
 
-            It 'should be $null' {
+            It 'should throw an error when greater than 128' {
                 $Splat = @{
-                    IPAddress = '10.0.0.2'
+                    IPAddress = 'fe80::1'
                     InterfaceAlias = 'Ethernet'
+                    SubnetMask = 129
+                    AddressFamily = 'IPv6'
                 }
-                $Result = Test-Properties @Splat -Apply
-                $Result | Should BeNullOrEmpty
+                { Test-ResourceProperty @Splat } | Should Throw
+            }
+            It 'should throw an error when less than 0' {
+                $Splat = @{
+                    IPAddress = 'fe80::1'
+                    InterfaceAlias = 'Ethernet'
+                    SubnetMask = -1
+                    AddressFamily = 'IPv6'
+                }
+                { Test-ResourceProperty @Splat } | Should Throw
             }
+        }
 
-            It 'should call all the mocks' {
-                Assert-MockCalled -CommandName Get-NetIPAddress
-                Assert-MockCalled -CommandName Get-NetConnectionProfile
-                Assert-MockCalled -CommandName Get-NetRoute
-                Assert-MockCalled -CommandName Get-NetIPInterface
-                Assert-MockCalled -CommandName Remove-NetRoute
-                Assert-MockCalled -CommandName Remove-NetIPAddress
-                Assert-MockCalled -CommandName New-NetIPAddress
-                Assert-MockCalled -CommandName Set-NetConnectionProfile
+        Context 'invoking with valid string IPv6 subnet mask' {
+
+            It 'should not throw an error' {
+                $Splat = @{
+                    IPAddress = 'fe80::1'
+                    InterfaceAlias = 'Ethernet'
+                    SubnetMask = '64'
+                    AddressFamily = 'IPv6'
+                }
+                { Test-ResourceProperty @Splat } | Should Not Throw
             }
         }
     }
 }
 
-# Cleanup after the test
-Remove-Item -Path $moduleRoot -Recurse -Force
```

Comment:
Actually, there isn't any reason! I think what might have happened is that I copied the header code xFirewall.Tests code you added in the recent version but didn't copy the footer cleanup code and therefore when I committed it removed it. I'll add it to mine as it should be there - always best to clean up afterwards :)